### PR TITLE
Add more DirectoryServices tests - 3

### PR DESCRIPTION
**Pull request by tarekgh**
This is adding some search and negative tests

### CODE REVIEW

**Comment by tarekgh** [comment]
CC @AlexGhiondea @hughbe @karelz @danmosemsft

**Comment by tarekgh** [comment]
@tquerec next, I am going to add the test for attributes and com objects. I have started that but ran into some problems with it. I'll investigate more and will contact you if I couldn't figure out solutions. thanks for your help.

**Comment by tarekgh** [comment]
#24316
#24388

File: src/System.DirectoryServices/tests/System/DirectoryServices/DirectoryServicesTests.cs [hunk 1]
```diff
@@ -303,6 +304,175 @@ public void TestDeleteTree()
             }
         }
 
+        [ConditionalFact(nameof(IsLdapConfigurationExist))]
+        public void TestDirectoryEntryNegativeCases()
+        {
+            // Test invalid server path
+            using (DirectoryEntry de = new DirectoryEntry("SomeWrongPath"))
```
Comment (danmosemsft) [inline src/System.DirectoryServices/tests/System/DirectoryServices/DirectoryServicesTests.cs]: can you break these into separate tests? doesn't seem like they need to be in the same method.

**Comment by WinCPP** [comment]
@tarekgh thank you for including me. I'm sorry, I could not figure out what to pick up... is there some carved out API set or work that I could start working on...?

**Comment by tarekgh** [comment]
Test Linux x64 Release Build please

File: src/System.DirectoryServices/tests/System/DirectoryServices/DirectoryServicesTests.cs (hunk 1 above)
Comment (tarekgh) [reply src/System.DirectoryServices/tests/System/DirectoryServices/DirectoryServicesTests.cs]: @danmosemsft could you please clarify why you think these are not different test cases and need to split into different methods? All these cases are testing the negative cases (the cases which throw exceptions) with the wrong usage.  I am seeing all these cases are related but I am open to split it if you can tell more details why you think so. Thanks.

**Comment by tarekgh** [comment]
@WinCPP thanks for your willing to help here. I am still exploring the area and trying to add basic tests which cover the main scenarios. I would suggest to you first try to get familiar how the test is working. you may look at the comment on the file https://github.com/dotnet/corefx/blob/master/src/System.DirectoryServices/tests/LDAP.Configuration.xml for soem info. 

I designed the test to be skipped if there is no define server configuration. And will run only if there is a server we can test against. you may try to run the tests on your side and when I come to the new namespaces, I'll try to let you know what areas you may start adding tests for. note, currently I am internally testing against 2 servers configureation, Windows server and Linux machine with OpenDJ to ensure I capture all different behaviors and ensure the test is reliable enough when running aginst different configureations. 

Thanks again for your willing to help here.

File: src/System.DirectoryServices/tests/System/DirectoryServices/DirectoryServicesTests.cs (hunk 1 above)
Comment (tarekgh) [reply src/System.DirectoryServices/tests/System/DirectoryServices/DirectoryServicesTests.cs]: I have split it into smaller methods. please let me know if you have any other comment.

**Comment by tarekgh** [comment]
test OSX x64 Debug Build please

**Comment by tarekgh** [comment]
@dotnet/dnceng could you please check OSX x64 Debug Build failure?

**Comment by chcosta** [comment]
@mmitche , has this issue already been dealt with?  How are you handling these cases when you see them?  Disable offending machines?  Is there additional follow-up?

```
09:04:31 Cloning repository https://github.com/dotnet/corefx
09:04:31  > git init /Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest # timeout=10
09:04:31 ERROR: Error cloning remote repo 'origin'
09:04:31 hudson.plugins.git.GitException: Could not init /Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest

09:04:31 stderr: xcrun: error: invalid active developer path (/Library/Developer/CommandLineTools), missing xcrun at: /Library/Developer/CommandLineTools/usr/bin/xcrun```

**Comment by tarekgh** [comment]
@tquerec please have a look when you have a chance. Thanks

**Comment by WinCPP** [comment]
@tarekgh Visited the OpenDJ website as per the instructions ins LDAP.Configuration.xml file. The website asks me to register giving company name. Found it a bit confusing as to what would be the legal implications for trial use as well.

I have got active directory installation on my local setup. Does this same config file work for AD as well?

Thanks!

**Comment by tarekgh** [comment]
> I have got active directory installation on my local setup. Does this same config file work for AD as well?

Yes, as long as you provide the configuration info (e.g server name, port, domain) it should work.

**Comment by WinCPP** [comment]
Hi @tarekgh, I have got ADLDS instance installed on my Windows 10 desktop. I am trying to get correct configuration of the same. I have edited the configuration.xml. But with this the tests are failing. Is this correct config? Please note that my home PC is a Windows 10 machine in work group; essentially no domain. Appreciate your inputs and if there are some steps to get this working on a desktop.

```xml
<?xml version="1.0" encoding="UTF-8"?>
<Configuration>
    <ServerName>localhost</ServerName>
    <Domain></Domain>
    <Port>389</Port>
    <User>cn=Directory Manager</User>
    <Password>password</Password>
    <AuthenticationTypes>ServerBind,None</AuthenticationTypes>
</Configuration>
```

Also are there any specific steps to create a particularly named instance using ADSI Edit?

Thanks,
Mandar

**Comment by tarekgh** [comment]
@WinCPP I I believe you have to have a Windows Server installed with Active Directory. ADLDS you have installed on the client is just to manage the AD on the server.
If you cannot install a Windows Server with Active Directory, then OpenDJ would be the other way to install it on the Windows client. 

CC @tquerec who is the expert here and can advise better regarding the needed configuration.

**Pull request merged**